app/src/main/java/org/wikipedia/feed/suggestededits/SuggestedEditsCardView.kt

```diff
@@ -0,0 +1,143 @@
+    interface Callback {
+        fun onSuggestedEditsCardClick(@NonNull pageTitle: PageTitle, @NonNull sourceDescription: String, @NonNull sourceLangCode: String, @NonNull view: SuggestedEditsCardView)
+    }
+
+    private val disposables = CompositeDisposable()
+    val CARD_BOTTOM_PADDING = 16.0f
+    val ARTICLE_EXTRACT_MAX_LINE_WITHOUT_IMAGE = 6
+    var translation: Boolean = false
+    private var sourceDescription: String = ""
+    private val app = WikipediaApp.getInstance()
+    private var summary: RbPageSummary? = null
+
+    init {
+        View.inflate(getContext(), R.layout.fragment_add_title_descriptions_item, this)
+    }
+
+    override fun setCard(@NonNull card: SuggestedEditsCard) {
+        super.setCard(card)
+        prepareViews()
+        translation = card.isTranslation()
+        summary = card.getsummary()
+        sourceDescription = card.getSourceDescription()
+        setLayoutDirectionByWikiSite(card.wikiSite(), rootView!!)
+        header(card)
+        updateSourceDescriptionWithHighlight()
+    }
+
+    private fun updateSourceDescriptionWithHighlight() {
+        if (translation) {
+            viewArticleSubtitleContainer.visibility = View.VISIBLE
+            viewArticleSubtitleAddedBy.visibility = View.GONE
+            viewArticleSubtitleEdit.visibility = View.GONE
+            viewArticleSubtitle.text = sourceDescription
+        }
+        updateContents()
+    }
+
+
+    override fun setCallback(callback: FeedAdapter.Callback?) {
+        super.setCallback(callback)
+        headerView.setCallback(callback)
+    }
+
+    private fun updateContents() {
+        viewAddDescriptionButton.visibility = View.VISIBLE
+        viewArticleTitle.text = summary!!.normalizedTitle
+        callToActionText.text = if (translation) String.format(context.getString(R.string.add_translation), app.language().getAppLanguageCanonicalName(app.language().appLanguageCodes.get(1))) else context.getString(R.string.editactionfeed_add_description_button)
+        showImageOrExtract()
+    }
+
+    private fun showImageOrExtract() {
+        if (TextUtils.isEmpty(summary!!.thumbnailUrl)) {
+            viewArticleImage.visibility = View.GONE
+            viewArticleExtract.visibility = View.VISIBLE
+            divider.visibility = View.VISIBLE
+            viewArticleExtract.text = StringUtil.fromHtml(summary!!.extractHtml)
+            viewArticleExtract.maxLines = ARTICLE_EXTRACT_MAX_LINE_WITHOUT_IMAGE
+        } else {
+            viewArticleImage.visibility = View.VISIBLE
+            viewArticleExtract.visibility = View.GONE
+            divider.visibility = View.GONE
+            viewArticleImage.loadImage(Uri.parse(summary!!.thumbnailUrl))
+        }
+    }
+
+    private fun prepareViews() {
+        viewArticleExtract.text = ""
+        viewArticleTitle.text = ""
+        callToActionText.text = ""
```

Comment:
Not sure why to set empty strings to the views...

Reply:
Done

Reply:
Sorry but I don't see they have been updated.

Reply:
Oh sorry ... Sometimes when it takes a bit of time to populate data in the cardView, the previous value would get shown because of recycling.. hence this..